### PR TITLE
tsp, avs, rename List API response schema back to ##List

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
No REST API change.

This would only affect SDK (maybe REST API doc).
Rename these model back to what's in prior stable api-versions, e.g.
https://github.com/Azure/azure-rest-api-specs/blob/c1416ca8c016762006ef928aca382c4fa0b26a26/specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/vmware.json#L425

---

I did not modify `OperationListResult`, as it is via interface template. Seems better to keep it.
https://github.com/Azure/azure-rest-api-specs/blob/c1416ca8c016762006ef928aca382c4fa0b26a26/specification/vmware/Microsoft.AVS/routes.tsp#L15

# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Other, please clarify:
    - Compatibility fix for SDK (fix issue introduced when service migrated to TypeSpec), no REST API change

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Review by cataggar — APPROVED** [review]

File: specification/vmware/Microsoft.AVS/routes.tsp [hunk 1]
```diff
@@ -658,7 +709,10 @@ interface WorkloadNetworkDnsServices {
 interface WorkloadNetworkDnsZones {
   #suppress "@azure-tools/typespec-azure-core/no-operation-id" "Can not change existing operationId."
   @operationId("WorkloadNetworks_ListDnsZones")
-  list is ArmResourceListByParent<WorkloadNetworkDnsZone>;
+  list is ArmResourceListByParent<
+    WorkloadNetworkDnsZone,
+    Response = ArmResponse<ResourceList<WorkloadNetworkDnsZone, "{name}sList">>
```
Comment (qiaozha) [inline specification/vmware/Microsoft.AVS/routes.tsp:714]: maybe a minor concern, is this the right way to get plural if the name is policy?

File: specification/vmware/Microsoft.AVS/routes.tsp [hunk 2]
```diff
@@ -38,9 +38,15 @@ interface Locations {
 interface PrivateClouds {
   #suppress "@azure-tools/typespec-azure-core/no-operation-id" "Can not change existing operationId."
   @operationId("PrivateClouds_List")
-  list is ArmResourceListByParent<PrivateCloud>;
+  list is ArmResourceListByParent<
```
Comment (qiaozha) [inline specification/vmware/Microsoft.AVS/routes.tsp:41]: will the naming fix be covered in 0.57.0? if yes, we will need to regen after typespec is released for this right?

File: specification/vmware/Microsoft.AVS/routes.tsp [hunk 3]
```diff
@@ -854,7 +920,13 @@ interface VirtualMachines {
 interface PlacementPolicies {
   #suppress "@azure-tools/typespec-azure-core/no-operation-id" "Can not change existing operationId."
   @operationId("PlacementPolicies_List")
-  list is ArmResourceListByParent<PlacementPolicy>;
+  list is ArmResourceListByParent<
+    PlacementPolicy,
+    Response = ArmResponse<ResourceList<
+      PlacementPolicy,
+      "PlacementPoliciesList"
```
Comment (qiaozha) [inline specification/vmware/Microsoft.AVS/routes.tsp:927]: oh I see we can specify the full name here.

**Review by qiaozha — APPROVED** [review]

**Review by cataggar — APPROVED** [review]
Thanks @weidongxu-microsoft.

**Review by raych1 — APPROVED** [review]

File: specification/vmware/Microsoft.AVS/routes.tsp (hunk 2 above)
Comment (weidongxu-microsoft) [reply specification/vmware/Microsoft.AVS/routes.tsp:41]: For this release, we likely would generate from Swagger. And operationId in Swagger is specified on above `@operationId("PrivateClouds_List")` line.

We are renaming these List in another PR (in client.tsp) https://github.com/Azure/azure-rest-api-specs/pull/29308
But as we know, the problem is how to rename resource group...

**Pull request merged**